base/bucket.go

```diff
@@ -128,6 +179,98 @@ func (b BucketSpec) GetViewQueryTimeout() time.Duration {
 
 }
 
+// TLSConnect method is passed to cbdatasource, to be used when creating a TLS-enabled memcached connection.
+// Establishes a connection, then wraps w/ gomemcached Client for use by cbdatasource.
+// Will include client cert (x.509) authentication when specified in the BucketSpec.
+func (b BucketSpec) TLSConnect(prot, dest string) (rv *memcached.Client, err error) {
+
+	d := net.Dialer{
+		Deadline: time.Now().Add(30 * time.Second),
+	}
+
+	host, port, err := SplitHostPort(dest)
+	if err != nil {
+		return nil, err
+	}
+
+	port = fmt.Sprintf("%d", b.KvTLSPort)
+	dest = host + ":" + port
+
+	Infof(KeyAll, "Establishing TLS connection for DCP to destination %s", dest)
+
+	conn, err := d.Dial("tcp", dest)
+	if err != nil {
+		return nil, err
+	}
+
+	tcpConn, isTcpConn := conn.(*net.TCPConn)
+	if !isTcpConn {
+		return nil, fmt.Errorf("Unable to convert connection to TCPConn during DCP TLS connection (Connection type:%T)", conn)
+	} else {
+		err = tcpConn.SetNoDelay(false)
```

Comment:
Can you add a comment that explains why `tcpConn.SetNoDelay()` is being set to `false`?  I think it's worth justifying why not to use the default setting here.

Reply:
I'm following gocb's prior art in this case. I've added a comment indicating that at the top of the method.

Reply:
That works.